File: Easy/746_MinCostClimbingStairs/Java/Solution.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int dp, dp_1 = 0, dp_2 = 0;

        for (int i = 2; i < cost.length + 1; i++) {
            dp = Math.min(dp_2 + cost[i - 2], dp_1 + cost[i - 1]);
            dp_2 = dp_1;
            dp_1 = dp;
        }

        return dp_1;
    }
}

File: Easy/746_MinCostClimbingStairs/C++/Solution.cpp
#include <vector>

class Solution {
public:
    int minCostClimbingStairs(std::vector<int>& cost) {
        int l = cost.size();
        int dp[l];
        dp[l - 1] = cost[l - 1];
        dp[l - 2] = cost[l - 2];

        for(int i = l - 3; i >= 0; i--) dp[i] = std::min(dp[i + 1], dp[i + 2]) + cost[i];

        return std::min(dp[0], dp[1]);
    }
};